utils/system_locale/src/linux.rs

```diff
@@ -0,0 +1,79 @@
+#[cfg(target_os = "linux")]
+mod linux_locale {
+    use locale::{
+        self,
+        linux::ffi::{
+            LC_ADDRESS, LC_ALL, LC_COLLATE, LC_CTYPE, LC_IDENTIFICATION, LC_MEASUREMENT,
+            LC_MESSAGES, LC_MONETARY, LC_NAME, LC_NUMERIC, LC_PAPER, LC_TELEPHONE, LC_TIME,
+        },
+    };
+    use std::{
+        collections::{HashMap, HashSet},
+        ffi::CStr,
+        ptr::null,
+    };
+
+    pub unsafe fn fetch_locale_settings() -> HashMap<i32, String> {
```

Comment:
this method should be either be safe, or private. If safe, any unsafeness should be encapsulated inside. If unsafe, it should have a comment stating the requirements for it to be safe.

Reply:
I'd suggest abstracting the `LC_*` to an enum and have this return `HashMap<THIS_ENUM, Locale>`

Reply:
agreed